src/Processors/QueryPlan/ReadFromMergeTree.cpp

```diff
@@ -1851,8 +1851,10 @@ bool ReadFromMergeTree::requestOutputEachPartitionThroughSeparatePort()
 
 ReadFromMergeTree::AnalysisResult ReadFromMergeTree::getAnalysisResult() const
 {
-    auto result_ptr = analyzed_result_ptr ? analyzed_result_ptr : selectRangesToRead();
-    return *result_ptr;
+    if (!analyzed_result_ptr)
```

Comment:
Out of curiosity I've been investigating and found that the only situation where we actually pass this value in the constructor (not nullptr) is when we use projections (generated in analyzeProjectionCandidate). The rest of the time it's empty on creation, so this change should help reduce the number of calls.